### PR TITLE
enable test coverage and coveralls.io

### PR DESCRIPTION
**Pull request by bjlittle**
This PR turns-on the `pytest` coverage. 

In addition, it also provides coverage of the Cython `.pyx`... which took a but of figuring out.

If you enable `coveralls.io` for the repo, you can get analysis of the coverage, but unfortunately `coveralls.io` is not picking up the `*.pyx`, which is a shame, but `pytest-cov` appears to be working just fine.

### CODE REVIEW

**Comment by bjlittle** [comment]
Okay, so I need to rebase first...

File: test/test_cftime.py [hunk 1]
```diff
@@ -686,10 +686,6 @@ def __init__(self, start, n, step, units, calendar='standard'):
             and `step` define the starting date, the length of the array and
             the distance between each date (in units).
 
-            :Example:
-            >>> t = TestTime(datetime(1989, 2, 18), 45, 6, 'hours since 1979-01-01')
-            >>> print num2date(t[1], t.units)
-            1989-02-18 06:00:00
```
Comment (bjlittle) [inline test/test_cftime.py]: This test doc-test is failing after I enabled runnable doc-tests, took the decision to remove it... hope that's okay?

**Comment by bjlittle** [comment]
Getting some appveyor failures... :disappointed: 

Ahhh, okay, so I also need to update the `.appveyor.yml` file to keep it inline with the `.travis.yml` updates.

**Comment by bjlittle** [comment]
Hoo-rah! Fixed the appveyor barf... It's now lookin' dangerously good to me :smile:

**Comment by jswhit** [comment]
Very cool @bjlittle!  I will merge in a day or two if no one has any concerns.

File: setup.py [hunk 2]
```diff
@@ -1,38 +1,96 @@
...
+        for rpath, _, fnames in os.walk(CFTIME_DIR):
+            for fname in fnames:
+                _, ext = os.path.splitext(fname)
+                if ext in ('.pyc', '.pyo', '.c', '.so'):
+                    artifact = os.path.join(rpath, fname)
+                    if os.path.exists(artifact):
+                        print('clean: removing file {!r}'.format(artifact))
+                        os.remove(artifact)
+                    else:
+                        print('clean: skipping file {!r}'.format(artifact))
```
Comment (bjlittle) [inline setup.py]: This is just a developer convenience. I got tired of manually cleaning out the Cython binary files, so now all the developer requires to do is::

```
$ python setup.py clean_cython
$ python setup.py build_ext --inplace
```

File: setup.py [hunk 3]
```diff
@@ -1,38 +1,96 @@
...
+    with open(fname, 'r') as fi:
+        result = ''.join(fi.readlines())
+    return result
+
+
+if FLAG_COVERAGE in sys.argv:
+    COMPILER_DIRECTIVES = {'linetrace': True,
+                           'warn.maybe_uninitialized': False,
+                           'warn.unreachable': False,
+                           'warn.unused': False}
```
Comment (bjlittle) [inline setup.py]: Turning on the Cython `linetrace` directive makes the building phase pretty noisy... this is my attempt to calm it down a wee bit by ignoring the verbose Cython based compiler warnings.

**Comment by bjlittle** [comment]
coveralls

It would also be pretty cool to get `coveralls.io` fully working.

At the moment it's ignoring the fact that we've got coverage of the much sought after `_cftime.pyx` file.

Perhaps, that can be tackled outside this PR.

**Comment by bjlittle** [comment]
Travis builds seems to have stopped...

**Comment by bjlittle** [comment]
@jswhit This all seems to be working now.

You'll need to go to https://coveralls.io/, sync your repos, activate coverage for `Unidata/cftime` and we should then see coveralls reporting the test coverage. Then add the badge to the `README.md`.

**Comment by jswhit** [comment]
I just activated cftime in coveralls.io

**Comment by jswhit** [comment]
Could you add the badge to README.md as part of this PR?

**Comment by bjlittle** [comment]
@jswhit As I'm not a member of the Unidata organisation, I don't have access to the badge on coveralls, but I've guessed the URL and added it to the `README.md`

**Comment by bjlittle** [comment]
@jswhit The travis  jobs succeeded and pushed coverage data to coveralls.io, could you check to see if coveralls.io has logged the data? I don't have access.

Also, appveyor fail, but it's not a test failure, rather a 500 gateway on their CI side... I can't restart the job to see it complete... can you? or merge with it as is? You call.

**Comment by bjlittle** [comment]
*sigh* See https://github.com/appveyor/ci/issues/58 ...

**Comment by jswhit** [comment]
coveralls.io says "916 of 1046 relevant lines covered (87.57%)" 
I added you as a collaborator to the cftime project on github, so you should be able to see the results on coveralls.io too.

**Comment by jswhit** [comment]
I restarted the appveyor build.

**Comment by bjlittle** [comment]
coveralls

@jswhit Awesome, thanks for the collaboration invite! :smile: 

I just checked coveralls.io and it seems good. Note that, you can browse the files covered an see what souce code has been hit (and how many times) by the tests, hence it's easy to identify gaps in the testing coverage

**Pull request merged**